### ISSUE TITLE
`--cache` flag not working

### ISSUE THREAD

**Issue by RyanMcCBing** [eslint/eslint#9802]
**Tell us about your environment**
It's just a simple testing setup. I made a new node package, installed eslint, `--init` with standard config, and dropped in a folder with a single .js file in it. The hierarchy is like this: `proj\test\test.js`

* **ESLint Version:** v4.14.0
* **Node Version:** v6.2.1
* **npm Version:** 3.9.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "extends": "standard"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I'm trying to use the `--cache` option, but I can't get it to work. I've also tried specifying a `--cache-location`, but it doesn't help.

```bash
C:\Users\RyanM\Desktop\proj>node_modules\.bin\eslint test -c .eslintrc.json -o out --cache
C:\Users\RyanM\Desktop\proj>node_modules\.bin\eslint test -c .eslintrc.json -o out --cache --cache-location .eslintcache
```

**What did you expect to happen?**
A `.eslintcache` file should be created in the CWD, or wherever I specify the cache-loaction.

**What actually happened? Please include the actual, raw output from ESLint.**
No `.eslintcache` file is created that I can find.

Here's the output when I add the `--debug` option
```
eslint:cli Running on files +0ms
eslint:config Using command line config .eslintrc.json +0ms
eslint:config-file Loading JSON config file: C:\Users\RyanM\Desktop\proj\.eslintrc.json +0ms
eslint:config-file Loading standard +295ms
eslint:config-file Attempting to resolve eslint-config-standard +2ms
eslint:config-file Loading JS config file: C:\Users\RyanM\Desktop\proj\node_modules\eslint-config-standard\index.js +1ms
eslint:glob-util Creating list of files to process. +0ms
eslint:ignored-paths Looking for ignore file in C:\Users\RyanM\Desktop\proj +0ms
eslint:ignored-paths Could not find ignore file in cwd +1ms
eslint:config Constructing config file hierarchy for C:\Users\RyanM\Desktop\proj\test +713ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Loading C:\Users\RyanM\Desktop\proj\.eslintrc.json +3ms
eslint:config Using C:\Users\RyanM\Desktop\proj\.eslintrc.json +1ms
eslint:config Using command line config file +1ms
eslint:config-ops Using config from partial cache +0ms
eslint:config-ops Apply environment settings to config +2ms
eslint:config-ops Creating config for environment es6 +1ms
eslint:config-ops Creating config for environment node +1ms
eslint:cli-engine Processing C:\Users\RyanM\Desktop\proj\test\test.js +0ms
eslint:cli-engine Linting C:\Users\RyanM\Desktop\proj\test\test.js +1ms
eslint:config Constructing config file hierarchy for C:\Users\RyanM\Desktop\proj\test +12ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using command line config file +1ms
eslint:config Using config from cache +0ms
eslint:linter Linting code for C:\Users\RyanM\Desktop\proj\test\test.js (pass 1) +0ms
eslint:linter Generating fixed text for C:\Users\RyanM\Desktop\proj\test\test.js (pass 1) +544ms
eslint:text-fixer Applying fixes +0ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Linting complete in: 589ms +552ms
```

**Comment by platinumazure** [comment]
Thanks for creating this issue, @RyanMcCBing. We'll investigate and see if we can reproduce.

**Comment by galvarez421** [comment]
@RyanMcCBing Do you get any warnings or errors when test.js is linted? Note that, according to [this comment](https://github.com/eslint/eslint/issues/2998#issuecomment-215239411), information for a linted file will be stored only if there are no warnings or errors found. So, if you are linting only a single file and there are warnings or errors for that file, then no .eslintcache file will be created. I ran into the same issue as you and the cause of the issue for me was the issue I just described. I don't think the documentation makes it clear that this is how caching behaves.

**Comment by RyanMcCBing** [comment]
@galvarez421 Ah, that's definitely the issue then. Thanks!

It would be nice if the CLI documentation mentioned this to avoid confusion in the future though.

**Comment by platinumazure** [comment]
I think this is a design gap. Issue #9948 was created to request that we store even files that failed linting in the cache.

@RyanMcCBing Would it be okay if we close this issue, and you can track #9948 for updates?

**Comment by platinumazure** [comment]
On second thought, maybe we could leave this open as a documentation issue, and we could accept a PR to correct the documentation so that it describes the current behavior. And then #9948 could be used for tracking an eventual enhancement to cache files that failed linting.

**Comment by stevenleija** [comment]
Any updates on this issue?  This would be very helpful as we are introducing linting to our code base so linting only changed files would be very helpful instead of seeing the 10,000+ errors/warnings we have each time we save a single file while watch is running.
